extensions/aws/processors/DeleteS3Object.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file DeleteS3Object.cpp
+ * DeleteS3Object class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DeleteS3Object.h"
+
+#include <set>
+#include <memory>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace aws {
+namespace processors {
+
+const core::Property DeleteS3Object::Version(
+  core::PropertyBuilder::createProperty("Version")
+    ->withDescription("The Version of the Object to delete")
+    ->supportsExpressionLanguage(true)
+    ->build());
+
+const core::Relationship DeleteS3Object::Success("success", "FlowFiles are routed to success relationship");
+const core::Relationship DeleteS3Object::Failure("failure", "FlowFiles are routed to failure relationship");
+
+void DeleteS3Object::initialize() {
+  // Set the supported properties
+  std::set<core::Property> properties(S3Processor::getSupportedProperties());
```

Comment:
Minor, but  instead of adding the properties manually and defining them twice`S3Processor` could set its supported properties in its constructor, and the interface for the `ConfigurableComponent` could be extended with insertion for new properties.

Reply:
Included in [1a59b59](https://github.com/apache/nifi-minifi-cpp/pull/931/commits/1a59b59b1141edaa82d45bae920f732c385e78fd). The only difference is that I did not introduce a new insertion operation in the `ConfigurableComponent` but used `updateSupportedProperties` instead.